src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -270,10 +273,59 @@ private void EmitResource(ResourceSymbol resourceSymbol)
 
             var typeReference = EmitHelpers.GetTypeReference(resourceSymbol);
             var body = resourceSymbol.DeclaringResource.Value;
-            if (body is IfConditionSyntax ifCondition)
+
+
+            var ancestors = this.context.SemanticModel.ResourceAncestors.GetAncestors(resourceSymbol);
+
+            var conditions = new List<SyntaxBase>();
+            foreach (var ancestor in ancestors)
             {
-                body = ifCondition.Body;
-                this.emitter.EmitProperty("condition", ifCondition.ConditionExpression);
+                if (ancestor.DeclaringResource.Value is IfConditionSyntax ifCondition)
+                {
+                    conditions.Add(ifCondition.ConditionExpression);
+                }
+            }
+
+            // Unwrap the 'real' resource body if there's a condition
+            {
+                if (body is IfConditionSyntax ifCondition)
+                {
+                    body = ifCondition.Body;
+                    conditions.Add(ifCondition.ConditionExpression);
+                }
+            }
+
+            if (conditions.Count == 1)
+            {
+                this.emitter.EmitProperty("condition", conditions[0]);
+            }
+            else if (conditions.Count > 1)
+            {
+                var @operator = new BinaryOperationSyntax(
+                    conditions[0], 
+                    SyntaxFactory.CreateToken(TokenType.LogicalAnd, "&"),
+                    conditions[1]);
+                for (var i = 2; i < conditions.Count; i++)
+                {
+                    @operator = new BinaryOperationSyntax(
+                        @operator,
+                        SyntaxFactory.CreateToken(TokenType.LogicalAnd, "&"),
+                        conditions[i]);
+                }
```

Comment:
Need test for more complex cases like this. Waiting on final design to get approved to see if we need it 😆